bench.sh

```diff
@@ -41,24 +42,29 @@ for benchmark in ${BENCHMARKS_TO_RUN}; do
 	fi
 
 	# Start the gRPC Server container
-	docker run --name "${NAME}" --rm \
+	docker run \
+		--name "${NAME}" \
+		--rm \
 		--cpus "${GRPC_SERVER_CPUS}" \
 		--memory "${GRPC_SERVER_RAM}" \
 		-e GRPC_SERVER_CPUS \
 		-e GRPC_SERVER_RAM \
-		--network=host --detach --tty "${NAME}" >/dev/null
+		--network=host \
+		--detach \
+		--tty \
+		"$GRPC_IMAGE_NAME:${NAME}-$GRPC_REQUEST_SCENARIO" >/dev/null
 
-	# Wait for server to be ready
-	sleep 5
+	printf 'Waiting for server to come up... '
+	while ! nc -z localhost 50051; do sleep .1; done
```

Comment:
Yes that one job will fail but only after a long time. So I'm adding a five minutes runtime timeout on this step.
Previously it would show zeroes but then the Ruby script would fail reading missing stats.

Reply:
@fenollp what about timeouts outside of CI? Like someone wanting to have a clean build and run all the benchmarks overnight?
 It won't be a beautiful one liner like with this `while` (I guess) but it may be more robust to implement this timeout here, what do you think?

Reply:
Oh you're right this should not spin forever. I'll add a counter and break on this.